numpy/core/src/multiarray/calculation.c

```diff
@@ -830,28 +830,38 @@ _GenericBinaryOutFunction(PyArrayObject *m1, PyObject *m2, PyArrayObject *out,
         return PyObject_CallFunction(op, "OO", m1, m2);
     }
     else {
-        PyObject *args, *kw, *ret;
+        PyObject *args, *ret, *unsafe;
+        static PyObject *kw = NULL;
 
-        args = Py_BuildValue("OOO", m1, m2, out);
-        if (args == NULL) {
-            return NULL;
-        }
-        kw = PyDict_New();
         if (kw == NULL) {
-            Py_DECREF(args);
-            return NULL;
+            kw = PyDict_New();
```

Comment:
Why not build the dictionary with `Py_BuildValue` and spare ourselves all the trouble? 

```C
kw = Py_BuildValue("{s:s}", "casting", "unsafe");
```